src/vs/code/node/cliProcessMain.ts

```diff
@@ -144,15 +143,14 @@ class CliMain extends Disposable {
 		fileService.registerProvider(Schemas.file, diskFileSystemProvider);
 
 		// State
-		const stateService = new StateService(environmentService, logService, fileService);
-		services.set(IStateService, stateService);
 
 		// Uri Identity
 		const uriIdentityService = new UriIdentityService(fileService);
 		services.set(IUriIdentityService, uriIdentityService);
 
 		// User Data Profiles
-		const userDataProfilesService = new UserDataProfilesService(stateService, uriIdentityService, environmentService, fileService, logService);
+		const stateService = new StateReadonlyService(SaveStrategy.DELAYED, environmentService, logService, fileService);
```

Comment:
I like this, I guess we were lucky before that no one did write to the state service from the CLI before when potentially an instance was already running.

However, I wonder if there is a user of state service from CLI that wants to write and is now broken, so maybe check.

Reply:
StateService was still readonly before except that naming was different. Readonly state service was called StateService and normal state service was called StateMainService. So we are good here

Reply:
Got it.